cirq/google/calibration/phased_fsim.py

```diff
@@ -365,6 +391,75 @@ def parse_result(self, result: CalibrationResult) -> PhasedFSimCalibrationResult
         """Decodes the characterization result issued for this request."""
 
 
+@json_serializable_dataclass(frozen=True)
+class XEBPhasedFSimCalibrationOptions(PhasedFSimCalibrationOptions):
+    """Options for configuring a PhasedFSim calibration using XEB.
+
+    XEB uses the fidelity of random circuits to characterize PhasedFSim gates. The parameters
+    of the gate are varied by a classical optimizer to maximize the observed fidelities.
+
+    Args:
+        n_library_circuits: The number of distinct, two-qubit random circuits to use in our
+            library of random circuits. This should be the same order of magnitude as
+            `n_combinations`.
+        n_combinations: We take each library circuit and randomly assign it to qubit pairs.
+            This parameter controls the number of random combinations of the two-qubit random
+            circuits we execute. Higher values increase the precision of estimates but linearly
+            increase experimental runtime.
+        cycle_min: We run the random circuits at a variety of cycle depths to fit an exponential
+            decay in the fidelity. This is the minimum number of cycles to execute for each
+            combination. The number of cycles is given as range(cycle_min, cycle_max, cycle_step)
+        cycle_max: The maximum number of cycles to execute for each combinations. The number of
+            cycles is given as range(cycle_min, cycle_max, cycle_step)
+        cycle_step: The increment in cycle depths. The number of cycles is given as
+            range(cycle_min, cycle_max, cycle_step).
+        fatol: The absolute convergence tolerance for the objective function evaluation in
+            the Nelder-Mead optimization. This controls the runtime of the classical
+            characterization optimization loop.
+        xatol: The absolute convergence tolerance for the parameter estimates in
+            the Nelder-Mead optimization. This controls the runtime of the classical
+            characterization optimization loop.
+        fsim_options: An instance of `XEBPhasedFSimCharacterizationOptions` that controls aspects
+            of the PhasedFSim characterization like initial guesses and which angles to
+            characterize.
+    """
+
+    n_library_circuits: int = 20
+    n_combinations: int = 10
+    cycle_min: int = 3
+    cycle_max: int = 100
+    cycle_step: int = 20
```

Comment:
I really don't think people should be executing non-equally-spaced-cycle-depths through the cloud API. Maybe people could experiment with different schemes using the open-source implementation and if there's a real need we can make it more flexible. My instinct is to keep the guard rails tight when using the cloud api and relax over time rather than try to do too much future proofing.

Practically, any adjustments you _would_ make are tweaks to these `arange` parameters so each non-default call would have to do a `tuple(range(..))` each time and the default argument would look funny in this class, as it would have to be written out

Reply:
we spoke offline and turns out exponential spacing is more common in practice. I've updated the code to use a tuple of ints